lib/ts/framework/utils.ts

```diff
@@ -136,17 +136,19 @@ export async function assertThatBodyParserHasBeenUsedForExpressLikeRequest(
             let jsonParser = json();
             let err = await new Promise((resolve) => {
                 let resolvedCalled = false;
-                /**
-                 * the setImmediate here is to counter the next.js issue
-                 * where the json parser would not resolve and thus the request
-                 * just hangs forever. Next.JS does json parsing on its own.
-                 */
-                setImmediate(() => {
-                    if (!resolvedCalled) {
-                        resolvedCalled = true;
-                        resolve(undefined);
-                    }
-                });
+                if (request.fromNextJS === true) {
+                    /**
+                     * the setImmediate here is to counter the next.js issue
+                     * where the json parser would not resolve and thus the request
+                     * just hangs forever. Next.JS does json parsing on its own.
+                     */
+                    setImmediate(() => {
+                        if (!resolvedCalled) {
+                            resolvedCalled = true;
+                            resolve(undefined);
+                        }
+                    });
+                }
```

Comment:
If it's next js, we don't need to parse anything right? So you can remove the setImmediate and the jsonParser if this if statmeent is true??

Reply:
nextjs do allow users to disable the default parser. In that case, we should be parsing the request body